### PR TITLE
Refresh Distributed test for HDFS

### PR DESCRIPTION
**Pull request by jakirkham**
Following up on [this comment]( https://github.com/dask/dask/pull/3607#discussion_r195173790 ).

Updates the testing style for HDFS using a Distributed Client. This makes it more consistent with similar tests using the Distributed Client elsewhere.

cc @mrocklin

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

File: dask/bytes/tests/test_hdfs.py [hunk 1]
```diff
@@ -297,8 +305,8 @@ def test_glob(hdfs):
             {basedir + p for p in ['/a', '/a1', '/a2', '/a3', '/b1', '/c', '/c2']})
 
 
+@pytest.mark.skipif(not distributed)
 def test_distributed(hdfs):
```
Comment (mrocklin) [inline dask/bytes/tests/test_hdfs.py]: Thanks for doing this.

I believe that this will also need the `loop` pytest fixture.

**Comment by mrocklin** [comment]
I don't think so.  `loop` is only necessary when using tornado (which is
started by dask.distributed).  I think that this is the only test where
it's relevant.

Also, in hopes of nerd sniping you, it would be nice to make all of this
more pleasant with pytest fixtures:
https://github.com/dask/distributed/issues/1966

Just in case you're interested :)

**Comment by jakirkham** [comment]
So flake8 doesn't seem to like the use of the `loop` variable. Any ideas?

```
$ flake8 dask/bytes/tests/test_hdfs.py 
dask/bytes/tests/test_hdfs.py:308:1: F811 redefinition of unused 'loop' from line 18
```

**Comment by mrocklin** [comment]
Oh, right, that's a pain

https://github.com/dask/distributed/blob/7909f27d591612537bb23974d089869edb11d097/distributed/tests/test_client.py#L50

File: dask/bytes/tests/test_hdfs.py [hunk 2]
```diff
@@ -297,8 +305,8 @@ def test_glob(hdfs):
             {basedir + p for p in ['/a', '/a1', '/a2', '/a3', '/b1', '/c', '/c2']})
 
 
-def test_distributed(hdfs):
-    dist = pytest.importorskip('distributed')
+@pytest.mark.skipif(not distributed)
```
Comment (jakirkham) [inline dask/bytes/tests/test_hdfs.py]: Seems the issue is `flake8` wants to complain about this line. Though this falls out of the function having the `loop` variable and a decorator.

**Comment by mrocklin** [comment]
Thanks @jakirkham

**Comment by jakirkham** [comment]
Thanks.

Appears `pytest` wants the `reason` filled out for the `skipif` as well. This caused the Python 3.4 build to fail. Have added this in PR ( https://github.com/dask/dask/pull/3627 ).

**Pull request merged**